### ISSUE TITLE
vless+http+reality config not working

### ISSUE THREAD

**Issue by EndermanCo** [MatsuriDayo/NekoBoxForAndroid#68]
Hello
This config doesn't work but it works in v2rayng:
vless://ARSHAM.6IX@www.mon3ter.ml:500?path=%2F&security=reality&encryption=none&pbk=c7qYCHxiCmzD5ayB8e6fNenpTiY1mJCAvpK1k_SXREw&fp=chrome&type=http&sni=github.io#Meow
vless://ARSHAM.6IX@70.34.208.112:500?security=reality&encryption=none&pbk=c7qYCHxiCmzD5ayB8e6fNenpTiY1mJCAvpK1k_SXREw&fp=chrome&type=http&sni=github.io#Meow
There was a bug in sing-box where the vless+http configuration didn't work if the host didn't exist, but it was solved and I don't think it's related to sing-box.
log:
[NB4A 6449372481741429000.log](https://github.com/MatsuriDayo/NekoBoxForAndroid/files/11131731/NB4A.6449372481741429000.log)

**Comment by H1JK** [comment]
sid can be empty, as [it was described](https://github.com/XTLS/REALITY):

> "shortIds": [ // 必填，客户端可用的 shortId 列表，可用于区分不同的客户端
                        "" // 若有此项，客户端 shortId 可为空
                        "0123456789abcdef" // 0 到 f，长度为 2 的倍数，长度上限为 16
                    ]

And I think it is also acceptable in sing-box: https://github.com/SagerNet/sing-box/commit/70cf681ff280bc4ced593bce7005bea60c583454

**Comment by EndermanCo** [comment]
> The `sid` can not be empty in sing-box. But it can be empty in Xray??
> 
> https://sing-box.sagernet.org/configuration/shared/tls/#short_id
> 
> As far as I remember, the "protocols" of these cores are not very compatible.

Yes, this is not required in xray, in sing-box, short_id is not required from 1.2-beta9 onwards.

**Comment by Ght55** [comment]
Very bad connected config vless/vmess/ trojan 
Pls update 
Not connected  cludeflare link?????

**Comment by EndermanCo** [comment]
Unfortunately, I tried to find a similar configuration, but I couldn't find it.(for test)
